transformedTests/readme.md

```diff
@@ -0,0 +1,13 @@
+Tests, for which the code has type information transformed in. This way, it's much easier to write mass tests for the validator.
+i.e.
+````typescript
+class A { // This class will be enhanced with type info
+    foo() { return true ? 123 : 'foo'; }
+}
+expect( reflect(A).getMethod('foo').returnType.isUnion() ).to.be.true
+````
+
+Currently, it uses the published typescript-rtti package from npm.
+
+TODO: Try to set it up as a monorepo (currently this fails because the code gets compiled to ../dist/transfomer and we get an error during the tspc run if we point the transformer to there. Propably there are some parent files missing. 
```

Comment:
You'll get an error if you haven't run `npm build` in the top directory first, since `dist` won't exist. If it's not that, then I'm not sure what issue you're running into. 

I'd like to hear more about what sort of mono-repo setup you're thinking of. We only really ship the one package, so a lerna-style `packages` folder seems a bit much.

Reply:
I thought of a NPM-workspaces (havent't used lerna yet). I was thinking of 
- Publishing the `transformer` and the `rtti` as 2 separate packages on NPM. So you can add the transformer as a  **dev**Dependency and safe space in your shipped application.
- ...something else which i don't remember so sharp currently: Something with source maps (but i've just had a quick look at the current project structure and this looks reasonable ) 

See my restfuncs project. I have a similar situation: Publishing a `server` (+`client` + `common`) and a `transformer` package. There are also [post-transformed integration tests](https://github.com/bogeeee/restfuncs/tree/main/tests/clientServer) (which reside under the root package. Somehow this ways the only way to get it working with jest)

Reply:
> You'll get an error if you haven't run npm build

No, the error i'm running into is:
```
TypeError: Cannot read properties of undefined (reading 'length')
    at isEmptyBlock (evalmachine.<anonymous>:114867:31)
    at emitBlock (evalmachine.<anonymous>:113038:28)
    at pipelineEmitWithHintWorker (evalmachine.<anonymous>:111752:20)
    at pipelineEmitWithHint (evalmachine.<anonymous>:111617:9)
    at pipelineEmitWithSourceMaps (evalmachine.<anonymous>:115709:7)
    at pipelineEmitWithComments (evalmachine.<anonymous>:115401:7)
    at pipelineEmit (evalmachine.<anonymous>:111566:7)
    at emit (evalmachine.<anonymous>:111534:7)
    at emitListItemWithParenthesizerRule (evalmachine.<anonymous>:115847:5)
    at emitNodeListItems (evalmachine.<anonymous>:114558:9)
```

I've had a quick comparison against the published binaries and my first thought/guess is a windows vs unix line ending difference (i.e. there's a string literal in the transformer code that has "\r\n" which is checked against). I'm running on linux. But i didn't investigate more. May be you can check that one.

Reply:
> You'll get an error if you haven't run npm build

it was the typescript version. Resolving...